### ISSUE TITLE
More blocking rules examples in README.md

### ISSUE THREAD

**Issue by billsargent** [webrecorder/browsertrix-crawler#80]
Could you provide a couple more examples of blockingRules on the README? 

I'm moving away from httrack and I'm having trouble figuring out how to block urls like https://www.dummyurl.com/gallery/* 

I've tried regex and just plain simple urls in a yaml...

```
blockRules:
- url: dummyurl.com/gallery/
  type: block
- url: /^https://www\.dummyurl\.com/gallery/[a-zA-Z]+$/i
  type: block
```
 
 But all I get after a few minutes of running the container is 
 
```
/app/node_modules/puppeteer-core/lib/cjs/puppeteer/common/assert.js:26
        throw new Error(message);.361
              ^

Error: Request is already handled! 
    at Object.assert (/app/node_modules/puppeteer-core/lib/cjs/puppeteer/common/assert.js:26:15)
    at HTTPRequest.abort (/app/node_modules/puppeteer-core/lib/cjs/puppeteer/common/HTTPRequest.js:308:21)ihavenreborn.com/auth/login
    at BlockRules.handleRequest (/app/util/blockrules.js:97:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /app/util/blockrules.js:70:9
```
 
I'm positive this is something very basic that I'm not understanding. Any help would be appreciated.

**Comment by ikreymer** [comment]
Thanks for reporting, that is an odd error message, will look into that.

Since standard YAML doesn't support regexes, the rules should just be strings. Try:

```
blockRules:
- url: dummyurl.com/gallery/
  type: block
- url: "https://www\.dummyurl\.com/gallery/[a-zA-Z]+$"
  type: block
```

If you run with `--logging stats,debug`, you should also get a printout of how the blockRules have been parsed in JS.

**Comment by billsargent** [comment]
Is this a problem with the host server? 

```
oot@bill-HP-Pavilion-15-Notebook-PC:/home/bill# cat crawls/block.yaml | docker run -i -p 9037:9037 -v $PWD/crawls:/crawls/ webrecorder/browsertrix-crawler crawl --profile /crawls/profiles/profile.tar.gz --url https://www.dummyurl.com/ --screencastPort 9037 --behaviors autoscroll,autoplay,autofetch,siteSpecific --scopeType host --workers 2 --newContext window --text  --combineWarc true --collection testcollection --waitUntil load,networkidle0 --timeout 300 --config stdin --logging stats,debug
With Browser Profile: /crawls/profiles/profile.tar.gz
Seeds [
  ScopedSeed {
    url: 'https://www.dummyurl.com/',
    include: [ /^https:\/\/www\.dummyurl\.com\// ],
    exclude: [],
    scopeType: 'host',
    sitemap: false,
    allowHash: false,
    maxDepth: 99999
  }
]
URL Block Rules:
== Start:     2021-08-13 21:53:56.074
* Rule for URL Regex: /https:\/\/www.dummyurl.com\/gallery\/[a-zA-Z]+$/
    Type: block / 0 (100.00%), errors: 0 (0.00%)
    In Frame Regex: any 0 pages/second)
    Resource Type: any / 0.0% memory
== Workers:   0

Screencast Server started on: 9037
Skipping HEAD check https://www.dummyurl.com/, invalid status 404
Skipping HEAD check https://www.dummyurl.com/auth/login, invalid status 404 Start:     2021-08-13 21:53:56.074
Skipping HEAD check https://www.dummyurl.com/gallery/test, invalid status 404tart:     2021-08-13 21:53:56.074
URL Blocked/Aborted: https://www.dummyurl.com/gallery/test in frame https://www.dummyurl.com/), errors: 0 (0.00%)
URL Blocked/Aborted: https://www.dummyurl.com/gallery/test in frame https://www.dummyurl.com/6.2% memory
/app/node_modules/puppeteer-core/lib/cjs/puppeteer/common/assert.js:26
        throw new Error(message);eborn.com/gallery/test
              ^

Error: Request is already handled!
    at Object.assert (/app/node_modules/puppeteer-core/lib/cjs/puppeteer/common/assert.js:26:15)
    at HTTPRequest.abort (/app/node_modules/puppeteer-core/lib/cjs/puppeteer/common/HTTPRequest.js:308:21)
    at BlockRules.handleRequest (/app/util/blockrules.js:97:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /app/util/blockrules.js:70:9

```

**Comment by ikreymer** [comment]
Could you share a URL / crawl config where this is happening (perhaps over email?). There's a fix, but also want to understand why its failing to block a particular URL and which one, and why it might not be blockable.

**Comment by billsargent** [comment]
How can I send you an email?

**Comment by ikreymer** [comment]
ilya [at] webrecorder.net should work. thanks!

**Comment by ikreymer** [comment]
The issue was two-fold:
- Duplicate interception handlers were being added on the same page object, causing the 'Request is already handled' error (which was also not being caught).
- For this use case, it seems the goal is to use the scope rules rather than the page resource block rules.

Clarified the distinction between them in README, and the page resource block rules will only apply to page resources and not to top-level pages.

This is being addressed in #81

**Comment by ikreymer** [comment]
Addressed in 0.4.4 release:
- README includes additional examples for scope rules and block rules, and explains when to use which.
- The 'Request is already handled!' error should not happen anymore, was caused by adding the interception handler multiple times on the same page.

For excluding all pages that start with `dummyurl.com/gallery/`, a scope exclusion of `--exclude dummyurl.com/gallery/` should suffice. This is hopefully better explained in the README.
